packages/modules/Extension-Ads/src/contentScript/content.js

```diff
@@ -767,14 +767,48 @@ if ( typeof vAPI === 'object' && !vAPI.contentScript ) {
       return content;
     },
     extractAdData(frameId, elem = null) {
-      const { title, company, text } = this.extractTexts(frameId, elem);
+      const { title, text } = this.extractTexts(frameId, elem);
       const content = this.extractContent(frameId, elem);
 
-      return { title, company, text, content };
+      return { title, text, content };
     },
     extractTexts(frameId, element) {
-      // TODO: implement texts extractor
-      return { title: null, company: null, text: null };
+      let title = null;
+      let longestText = '';
+
+      if (!element) {
+        return { title: null, text: null };
+      }
+
+      const selectors = [
+        'h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'p', 'span',
+        'yt-formatted-string', 'body-text', 'title-text'
+      ];
+
+      const excludedWords = ['Right chevron', 'Like', 'Comment', 'Verified', 'Learn more', 'Sponsored', 'See Translation'];
+
+      const elements = Array.from(element.querySelectorAll(selectors)).filter((el) => {
+        const textContent = el.textContent.trim();
+
+        return !el.classList.contains('visually-hidden') &&
+          !excludedWords.some((word) => textContent.includes(word));
+      });
+
+      elements.forEach((el) => {
+        const textContent = el.textContent.trim();
+
+        if (textContent !== '') {
+          if (!title) {
+            title = textContent;
+          }
+          if (textContent.length > longestText.length) {
```

Comment:
Need to remove symbols from textContent before length check, but save textContent as it is in html.